server/src/main/java/org/elasticsearch/cluster/SimpleBatchedExecutor.java

```diff
@@ -71,7 +71,12 @@ public final ClusterState execute(BatchExecutionContext<Task> batchExecutionCont
                 Tuple<ClusterState, TaskResult> result = executeTask(task, clusterState);
                 clusterState = result.v1();
                 final var taskResult = result.v2();
-                taskContext.success(() -> taskSucceeded(task, taskResult));
+                Runnable successFunction = () -> taskSucceeded(task, taskResult);
+                if (task instanceof ClusterStateAckListener ackListenerTask) {
+                    taskContext.success(successFunction, ackListenerTask);
```

Comment:
This is to work around the assertion at https://github.com/elastic/elasticsearch/blob/80deeb81127656aaca70d62338278bdb0fdf3d31/server/src/main/java/org/elasticsearch/cluster/service/MasterService.java#L859

Reply:
Didn't spot that class but yes that works nicely :-) Have refactored it to use the new class and the ILM tests seem to be happy about it.